hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -124,9 +124,8 @@ def try_issubclass(thing, superclass):
                 return True
             for orig_base in getattr(thing, "__orig_bases__", None) or [None]:
                 args = getattr(orig_base, "__args__", None)
-                if not _compatible_args(args, superclass_args):
-                    return False
-            return True
+                if _compatible_args(args, superclass_args):
+                    return True
```

Comment:
I don't think this is correct - if we have multiple bases I think we want to check that the args are compatible for every base with args, not just that there is at least one base with compatible args.

Reply:
Ok, I see. Here's the code that does not work for me with this change: https://github.com/dry-python/returns/blob/1e7eeee2816dd8525e8596a83dc700648410025d/returns/interfaces/specific/reader_result.py#L102-L113

As you can see this class has 3 different parents with different number of type arguments and their type arguments are incompatible with each other on the first sight. However, it type-checks perfectly.

Reply:
Ah, makes sense. I'll re-review properly in the morning, but I think with this we're ready to merge?

Reply:
Yes, I think so too. Thanks a lot for your help! 👍